`Development`: Use id selectors in cypress e2e tests for text exercises

jhainzlschmid opened the pull request:
<!-- Thanks for contributing to Artemis! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
To improve the stability of the e2e test suite we want to only use id selectors to match UI elements.

### Description
<!-- Describe your changes in detail -->
In this PR I refactor the existing cypress tests for text exercises (and the used page objects) to not use css selectors or selectors which depend on a translation which comes from the Artemis Client.

### Steps for Testing
<!-- Please describe in detail how the reviewer can test your changes. -->
[The tests are executed by Bamboo and still pass](https://bamboo.ase.in.tum.de/browse/ARTEMIS-AETBBT-694)

### Review Progress
<!-- Each Pull Request should be reviewed by at least two other developers. The code as well as the functionality (= manual test) needs to be reviewed. -->
<!-- The reviewer or author check the following boxes depending on what was reviewed or tested. All boxes should be checked before merge. -->
<!-- You can add additional checkboxes if it makes sense to only review parts of the code or functionality. -->
<!-- When changes are pushed, uncheck the affected boxes. (Not all changes require full re-reviews.) -->

#### Code Review
- [x] Review 1
- [ ] Review 2

rs-maker commented on src/test/cypress/support/pageobjects/exercises/text/TextEditorPage.ts:
```diff
@@ -23,30 +23,22 @@ export class TextEditorPage {
     }
 
     getHeaderElement() {
         return cy.get('jhi-header-participation-page');
```
This isnt an id.

jhainzlschmid replied:
Correct this method isn't used in a test anymore, so I missed it

jhainzlschmid replied:
I removed it

rs-maker commented on src/test/cypress/support/pageobjects/exercises/text/TextExerciseCreationPage.ts:
```diff
@@ -55,8 +55,8 @@ export class TextExerciseCreationPage {
         cy.get('[jhitranslate="artemisApp.modelingExercise.createExampleSubmission"]').click();
```
This isnt an id either.

jhainzlschmid replied:
Same here the method isn't used anymore

jhainzlschmid replied:
I removed it

rs-maker reviewed (changes requested):
Mostly ok i left two comments where possible and mention the remaining selectors in the list below.

TextExerciseAssessmentPage.ts
- line 9: not an id but also not used at all. is this an artifact? 
TextEditorPage.ts:
- line 7: two selectors that are not ids
- line 15: two selectors that are not ids
- line 21: one selector that is not an id
- line 34: not an id

jhainzlschmid commented:
> Mostly ok i left two comments where possible and mention the remaining selectors in the list below.
> 
> TextExerciseAssessmentPage.ts
> 
> * line 9: not an id but also not used at all. is this an artifact?

Yes I removed it

> TextEditorPage.ts:
> 
> * line 7: two selectors that are not ids

Why two selectors? I replaced the one selector with an id.

> * line 15: two selectors that are not ids

This code is not used by the text exercise specs. Its only used in an exam related spec, so I will refactor it there

> * line 21: one selector that is not an id

Good catch, I missed it. Should be fine now

> * line 34: not an id

Do you mean the `contains` command? I replaced it with a direct assertion, but in some cases this might be more cumbersome than using `contains`

jhainzlschmid commented:
[Tests still pass after the changes](https://bamboo.ase.in.tum.de/browse/ARTEMIS-AETBBT-700)z

rs-maker reviewed (approved):
👍 As far as i can tell this is great now.
